autofs-5.1.0-beta1 - add plus to path match pattern

From: Ian Kent <raven@themaw.net>

The pattern matching in the tokenizer is very strict and quite
limited. That's been done (initially) to ensure that the parsing
of other parts of map entries is accurate and to ensure that
error cases are handled cleanly since the strict matching will
cause errors.

I don't want to open up patterns that probably need it to accept
a wide range of characters just yet but adding "+" to the match
is obviously needed.
---
 CHANGELOG         |    1 +
 modules/amd_tok.l |    2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/CHANGELOG b/CHANGELOG
index 70414d2..ed49983 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -214,6 +214,7 @@
 - fix ldap_uri config update.
 - fix typo in conf_load_autofs_defaults().
 - fix hash on confg option add and delete.
+- add plus to path match pattern.
 
 03/09/2009 autofs-5.0.5
 -----------------------
diff --git a/modules/amd_tok.l b/modules/amd_tok.l
index 618bc91..bb52747 100644
--- a/modules/amd_tok.l
+++ b/modules/amd_tok.l
@@ -81,7 +81,7 @@ V4NUM		([01]?[0-9][0-9]?|2[0-4][0-9]|25[0-5])
 MACRO		(\$\{([[:alpha:]_/]([[:alnum:]_\-])([[:alnum:]_\-/])*)\})
 QSTR		(\"([^"\\]|\\.)*\")
 OSTR		([[:alpha:]]([[:alnum:]_\-])+)
-FSTR		([[:alnum:]_/\.]([[:alnum:]_\-/\.]|(\\.))*)
+FSTR		([[:alnum:]_/\.]([[:alnum:]_\-/\+\.]|(\\.))*)
 VSTR		(([[:alnum:]_\-\:/\.])+)
 SSTR		([[:alpha:]]([[:alnum:]\-\.])+)
 IP4ADDR		({V4NUM}\.((({V4NUM}\.){0,2}){V4NUM}?))
